src/AbstractHydrator.php

```diff
@@ -215,83 +203,75 @@ public function getFilter()
      *
      * @param string $name Index in the composite
      * @param callable|Filter\FilterInterface $filter
-     * @param int $condition
-     * @return Filter\FilterComposite
      */
-    public function addFilter($name, $filter, $condition = Filter\FilterComposite::CONDITION_OR)
+    public function addFilter(string $name, $filter, int $condition = Filter\FilterComposite::CONDITION_OR) : void
     {
-        return $this->filterComposite->addFilter($name, $filter, $condition);
+        $this->filterComposite->addFilter($name, $filter, $condition);
     }
 
     /**
      * Check whether a specific filter exists at key $name or not
      *
-     * @param string $name Index in the composite
-     * @return bool
+     * @param string $name Index/name in the composite
      */
-    public function hasFilter($name)
+    public function hasFilter(string $name) : bool
     {
         return $this->filterComposite->hasFilter($name);
     }
 
     /**
      * Remove a filter from the composition.
-     * To not extract "has" methods, you simply need to unregister it
+     *
+     * To not extract "has" methods, unregister the filter.
      *
      * <code>
      * $filterComposite->removeFilter('has');
      * </code>
-     *
-     * @param $name
-     * @return Filter\FilterComposite
      */
-    public function removeFilter($name)
+    public function removeFilter(string $name) : void
     {
-        return $this->filterComposite->removeFilter($name);
+        $this->filterComposite->removeFilter($name);
     }
 
     /**
      * Adds the given naming strategy
      *
      * @param NamingStrategy\NamingStrategyInterface $strategy The naming to register.
-     * @return self
      */
-    public function setNamingStrategy(NamingStrategy\NamingStrategyInterface $strategy)
+    public function setNamingStrategy(NamingStrategy\NamingStrategyInterface $strategy) : void
     {
         $this->namingStrategy = $strategy;
-
-        return $this;
     }
 
     /**
      * Gets the naming strategy.
      *
-     * @return NamingStrategy\NamingStrategyInterface
+     * If no naming strategy is registered, registers the
+     * `IdentityNamingStrategy`, which acts essentially as a no-op.
+     *
+     * {@inheritDoc}
      */
-    public function getNamingStrategy()
+    public function getNamingStrategy() : NamingStrategy\NamingStrategyInterface
     {
+        if (null === $this->namingStrategy) {
+            $this->namingStrategy = new NamingStrategy\IdentityNamingStrategy();
+        }
         return $this->namingStrategy;
     }
 
     /**
      * Checks if a naming strategy exists.
-     *
-     * @return bool
      */
-    public function hasNamingStrategy()
+    public function hasNamingStrategy() : bool
     {
         return isset($this->namingStrategy);
```

Comment:
This should be replaced with `return null !== $this->namingStrategy`;

Reply:
They are semantically the same.